### PR TITLE
Try to add duration

### LINKED ISSUE #11: Duration data
Hello

Nice package but I have a question about duration:

Is there any duration data exposed in the GPX ?
like
- total duration
- moving duration
- cumulativeDuration (I see `distance.cumulative` so why not)

Thanks

### PR DESCRIPTION
**Pull request by Its-Just-Nans**
May fix https://github.com/We-Gold/gpxjs/issues/11

### CODE REVIEW

**Comment by Its-Just-Nans** [comment]
Apparently it could be better to apply threeshold with 10 seconds check


> https://support.strava.com/hc/en-us/articles/115001188684-Moving-Time-Speed-and-Pace-Calculations
> What constitutes a rest? Is it when you rest for 1, 3, 10, or 20 seconds? And what does your speed have to fall below for you to be considered at rest? One second might capture too many false positives, and 20 seconds may be too strict. One calculation isn't necessarily correct or incorrect, but we feel we use standards that most athletes would agree upon.

**Comment by We-Gold** [comment]
Can you add some comments to explain some of the calculations and constants in the `calculateDuration` function? I recognize that not all the other functions in _math_helpers_ are well commented, but in this case, it is important since you have some unexplained numbers. If someone looks at the code in the future, they may wonder why you divide by 1000 or have a threshold of exactly 0.00055.

The Strava example seems specific to GPX files from activity trackers (e.g. a GPS-enabled watch) since it assumes human movement. It's fine to calculate a thresholded version of the durations based on the Strava information, but that should be in addition to the current duration information that doesn't assume human movement.

In the future, it would be nice to add an option object to pass into the main `parseGPX` function, to allow a user to specify what they want calculated or maybe specific thresholds. Since that feature doesn't currently exist, I'd prefer to not make the code too opinionated, and anything opinionated should be in addition to the expected calculations.

**Comment by Its-Just-Nans** [comment]
I updated the code

File: lib/math_helpers.ts [hunk 1]
```diff
@@ -49,6 +49,91 @@ export const haversineDistance = (point1: Point, point2: Point): number => {
...
+						sumDistances +=
+							distance.cumulative[j + 1] - distance.cumulative[j]
+						sumTime += timeDiff
+					}
+				}
+
+				const avgSpeed = sumTime > 0 ? sumDistances / sumTime : 0
+
+				// Determine if average speed indicates resting
+				console.log(avgSpeed)
```
Comment (We-Gold) [inline lib/math_helpers.ts:100]: Remove console.log

File: lib/parse.ts [hunk 2]
```diff
@@ -11,25 +12,41 @@ import {
 	Track,
 	Waypoint,
 	Extensions,
+	Options,
 } from "./types"
 
+const DEFAULT_THRESHOLD = 0.000215 // m/ms - 0.215 m/s - 774000 km/h
```
Comment (We-Gold) [inline lib/parse.ts]: Move DEFAULT_THRESHOLD and DEFAULT_OPTIONS to a new file, like options.ts.

File: lib/parse.ts [hunk 3]
```diff
@@ -11,25 +12,41 @@ import {
...
  * Converts the given GPX XML to a JavaScript Object with the ability to convert to GeoJSON.
  *
  * @param gpxSource A string containing the source GPX XML
  * @param removeEmptyFields Whether or not to remove null or undefined fields from the output
  * @returns A ParsedGPX with all of the parsed data and a method to convert to GeoJSON
  */
-export const parseGPX = (gpxSource: string, removeEmptyFields: boolean = true) => {
+export const parseGPX = (
+	gpxSource: string,
+	removeEmptyFields: boolean = true
```
Comment (We-Gold) [inline lib/parse.ts]: Pass in options object to `parseGPX`, the same as with `parseGPXWithCustomParser`. Then merge this with the default options to fill in any unspecified values.

**Review by We-Gold — CHANGES_REQUESTED** [review]
Aside from the two specific comments, the options object needs to be documented in the readme. It needs its own section, so I suggest switching the old heading "Documentation" to "Types" and adding a new section above that section for "Options". Also, make sure to show an example of how to pass them in.

The Duration type and the changes to other types also need to be added to the types section of the readme documentation (currently called documentation).

File: README.md [hunk 4]
```diff
@@ -141,7 +141,18 @@ I do try to be responsive to PRs though, so if you leave one I'll try to get it
 
 Also, there are some basic tests built in to the library, so please test your code before you try to get it merged (_just to make sure everything is backwards compatible_). Use `npm run test` to do this.
 
-# Documentation
+## Options
+
+You can also run `parseGPX()` with custom options for more control over the parsing process. See [options.ts](./lib/options.ts) for more details.
```
Comment (We-Gold) [inline README.md:146]: The individual options need to be explained either in the README or the options.ts file you link to. If you do it in the file, comments on each line explaining each option should suffice for now.

**Comment by We-Gold** [comment]
Since the code looks good, I will test it locally later (hopefully today). If everything goes well, I can approve and merge it.

**Review by We-Gold — APPROVED** [review]
After testing locally, the calculations seem to be correct, and I didn't discover any errors. All the tests passed.

**Pull request merged**